content/docs/instrumenting/writing_exporters.md

```diff
@@ -5,67 +5,46 @@ sort_rank: 5
 custom collector. The theory covered will also be of interest to those doing
 direct instrumentation.
 
-If you are writing an exporter and are unclear on anything here, contact us on
-IRC (#prometheus on Freenode) or the [mailing list](/community).
+If you are writing an exporter and are unclear on anything here, please contact us on IRC (#prometheus on Freenode) or the [mailing list](/community).
 
 ## Maintainability and purity
 
 The main decision you need to make when writing an exporter is how much work
 you’re willing to put in to get perfect metrics out of it.
 
 If the system in question has only a handful of metrics that rarely change,
-then getting everything perfect is an easy choice (e.g. the [haproxy
-exporter](https://github.com/prometheus/haproxy_exporter)).
+then getting everything perfect is an easy choice, a good example of this is the [HAProxy exporter](https://github.com/prometheus/haproxy_exporter).
 
-If on the other hand the system has hundreds of metrics that change
-continuously with new versions, if you try to get things perfect then you’ve
-signed yourself up for a lot of ongoing work. The [mysql
-exporter](https://github.com/prometheus/mysqld_exporter) is on this end of the
+If, on the other hand, the system has hundreds of metrics that change
+frequently with new versions, if you try to get things perfect then you’ve
+signed yourself up for a lot of ongoing work. The [MySQL exporter](https://github.com/prometheus/mysqld_exporter) is on this end of the
 spectrum.
 
-The [node exporter](https://github.com/prometheus/node_exporter) is a mix,
-varying by module. For mdadm we have to hand-parse a file and come up with our
-own metrics, so we may as well get the metrics right while we’re at it. For
-meminfo on the other hand, the results vary across kernel versions so we end up
-doing just enough of a transform to create valid metrics.
+The [node exporter](https://github.com/prometheus/node_exporter) is a mix of these, with complexity varying by module. For example, the `mdadm` collector hand-parses a file and exposes metrics created specifically for that collector, so we may as well get the metrics right. For the `meminfo` collector the results vary across kernel versions so we end up doing just enough of a transform to create valid metrics.
 
 ## Configuration
 
 When working with applications, you should aim for an exporter that requires no
 custom configuration by the user beyond telling it where the application is.
 You may also need to offer the ability to filter out certain metrics if they
-may be too granular and expensive on large setups (e.g. the haproxy exporter
-allows filtering of per-server stats). Similarly there may be expensive metrics
-that are disabled by default.
+may be too granular and expensive on large setups, for example the [HAProxy exporter](https://github.com/prometheus/haproxy_exporter) allows filtering of per-server stats. Similarly there may be expensive metrics that are disabled by default.
```

Comment:
The line breaks are very inconsistent now.

Reply:
I'm going to fix the whole file.